ghost/core/core/server/data/migrations/versions/5.55/2023-07-10-05-16-55-add-built-in-collection-posts.js

```diff
@@ -0,0 +1,104 @@
+const logging = require('@tryghost/logging');
+const {default: ObjectID} = require('bson-objectid');
+const {chunk} = require('lodash');
+const {createTransactionalMigration} = require('../../utils');
+
+const insertPostCollections = async (knex, collectionId, postIds) => {
+    logging.warn(`Batch inserting ${postIds.length} collection posts for collection ${collectionId}`);
+
+    const collectionPosts = postIds.map((postId) => {
+        return {
+            id: (new ObjectID()).toHexString(),
+            collection_id: collectionId,
+            post_id: postId,
+            sort_order: 0
+        };
+    });
+
+    const batches = chunk(collectionPosts, 1000);
+
+    for (const batch of batches) { // eslint-disable-line no-restricted-syntax
```

Comment:
This part might cause performance issues. Needs additional checks on an instance with large amount of posts.

Reply:
Ran the migration on a db with ~4K records. Took `94.457ms` on my local machine. Imo we should be fine here. @daniellockyer any extra checks you'd perform for this kind of batch insert?

Reply:
For the scalability target of `100,000` the napkin math is 94ms x 25 = `2350ms` to run this query on local machine. If we assume the Pro environment is 10x slower (interesting metric to actually find out maybe?) that would be ~23s or running time.

Reply:
FWIW you don't need to loop over chunks because that's what `batchInsert` does: https://knexjs.org/guide/utility.html#batchinsert

Reply:
TIL! Ty